content/docs/tech-insights/track-sonarcloud/index.md

```diff
@@ -0,0 +1,145 @@
+
+1. First generate a new token in the Sonarcloud dashboard
+Click on your profile in the top right of the application, click My Account. Then click Security. Enter a token name and click "Generate Token". Copy the token value.
+2. In Roadie click Administration -> Settings -> Secrets and save the token to the secret named SONARCLOUD_API_TOKEN.
+
+## Configure a proxy in backstage so that it can communicate with the Sonarcloud API
+
+1. In Roadie visit Administration -> Settings -> Proxy and click "Add Proxy"
+2. Set the path of the new proxy to `/sonarcloud.io/api`
+3. Set the target to `https://sonarcloud.io/api`
+4. Expand the advanced section and click "Add" under the Headers section
+5. Enter "Authorization" for the header name and "Bearer ${SONARCLOUD_API_TOKEN}" as the header value
+6. In the Allow Headers section click "Add Item" and set its value to "Authorization"
+7. Lastly, click the secure checkbox to make sure that it uses the HTTPS protocol.
+
+## Create an HTTP Data Source to collect issues from Sonarcloud
+
+1. Within Roadie, click on Tech Insights -> Data Sources and click "Add Datasource"
+2. Set the name to "Sonarcloud Issues", and set the description if you prefer.
+3. Select the HTTP Data Provider, then select the proxy we created earlier.
+4. In the Path Extension enter `issues/search?projectKeys={{metadata.annotations["sonarqube.org/project-key"]}}`
+5. Next select an entity that you know has been configured with the Sonarcloud annotation so that we can test the data extraction in the next step.
+6. Click view. You should see the HTTP response from Sonarcloud with counts of issues like you see below:
+
+![Edit Data Source](edit-datasource.png)
+
+7. Next you can select the fields to extract from the API.
+In the Field extraction section select the "JSON with JSONata syntax" extractor. Then enter "count" as the fact name, `$count($.issues[status != "CLOSED"])` as the JSONata query, and "Integer" type. Add a second fact by clicking "Add Fact", enter "debtTotal", `$.debtTotal` and "Integer"
+
+8. Test the Data Source fact extraction by clicking "Check Facts". You should see something like the following.
+![Field Extraction](field-extraction.png)
+
+9. In the Applies to selection, set the "Has Annotation" field to "sonarqube.org/project-key"
+10. Click Save.
+11. In the Data Source page, click "Actions" and "Run" to trigger an initial run of the Data Source
+
+## Create an HTTP Data Source to collect hotspots from Sonarcloud
+
+1. Within Roadie, click on Tech Insights -> Data Sources and click "Add Datasource"
+2. Set the name to "Sonarcloud Hotspots", and set the description if you prefer.
+3. Select the HTTP Data Provider, then select the proxy we created earlier.
+4. In the Path Extension enter `hotspots/search?projectKey={{metadata.annotations["sonarqube.org/project-key"]}}`
+5. Next select an entity that you know has been configured with the Sonarcloud annotation so that we can test the data extraction in the next step.
```

Comment:
```suggestion
5. Next select an entity that you know has been configured with the SonarCloud annotation so that we can test the data extraction in the next step.
```